packages/fx-core/src/plugins/solution/fx-solution/v2/deploy.ts

```diff
@@ -197,15 +199,38 @@ export async function deploy(
   ctx.logProvider.info(getLocalizedString("core.deploy.startNotice", PluginDisplayName.Solution));
   const result = await executeConcurrently(thunks, ctx.logProvider);
 
+  const botTroubleShootLink =
+    "https://aka.ms/teamsfx-bot-help#how-can-i-troubleshoot-issues-when-teams-bot-isnt-responding-on-azure";
+  const botTroubleShootDesc = getLocalizedString("core.deploy.botTroubleShoot");
+  const botTroubleShootLearnMore = getLocalizedString("core.deploy.botTroubleShoot.learnMore");
   if (result.kind === "success") {
     if (inAzureProject) {
-      const msg = getLocalizedString("core.deploy.successNotice", ctx.projectSetting.appName);
-      ctx.logProvider.info(msg);
-      ctx.userInteraction.showMessage("info", msg, false);
+      let msg = getLocalizedString("core.deploy.successNotice", ctx.projectSetting.appName);
+      // Append a new sentence for bot trouble shootting.
+      if (inBotProject) {
+        msg += ` ${botTroubleShootDesc}`;
+      }
+      ctx.logProvider.info(msg + ` ${botTroubleShootLearnMore}: ${botTroubleShootLink}.`);
+      if (inBotProject) {
+        // Show a `Learn more` action button for bot trouble shooting.
+        ctx.userInteraction
+          .showMessage("info", msg, false, botTroubleShootLearnMore)
+          .then((result) => {
+            const userSelected = result.isOk() ? result.value : undefined;
+            if (userSelected === botTroubleShootLearnMore) {
+              ctx.userInteraction.openUrl(botTroubleShootLink);
+            }
+          });
+      } else {
+        ctx.userInteraction.showMessage("info", msg, false);
+      }
     }
     return ok(Void);
   } else {
-    const msg = getLocalizedString("core.deploy.failNotice", ctx.projectSetting.appName);
+    let msg = getLocalizedString("core.deploy.failNotice", ctx.projectSetting.appName);
```

Comment:
I am a little confused about the user experience, why it doesn't show dialog when deploy fails?

Reply:
Although I'm not the owner to answer the question, I guess a specific error will popped up.